sympy/solvers/solveset.py

```diff
@@ -978,6 +989,349 @@ def _solveset(f, symbol, domain, _check=False):
+    equation is identified of being a particular type.
+
+
+    How to add new class of equations
+    =================================
+
+    `\_transolve` is designed in such a way that it becomes an easy task to
+    add a new class of equation solver.
+
+    The first task that needs to be done for adding your own solver
+    is to decide from where its `identification helper` will be
+    invoked. To do so determine the general form of the class of the
+    equation to place the invocation of the identification helper to
+    an appropriate place within \_transolve, for example the general
+    form of the exponential equation is `a*f(x) + b*g(x)` where
+    `f(x)` and `g(x)` are power (or log) terms so we need to place the
+    invocation condition inside `Add` case. Once the place for
+    identification helper is determined, you can add a call to solving
+    helper. For your class of equation you need to define your own
+    identification and solving helpers. The identification helper
+    should be implemented for generalised cases and should return
+    either `True` if the given equation belongs to the class otherwise
+    `False`. Solving helpers needs to be implemented with such
+    heuristics or algorithms that solves most of the equations belonging
+    to that class. The value returned from this helper should be handled
+    properly in the main (\_transolve) function itself. It could be either
+    the exact solution or a reduced form of the equation which
+    `solveset` can handle.
+
+    Apart from this, few other things needs to be taken care while adding
+    an equation solver:
+
+    - Naming conventions:
+      Name of the identification helper should be like  `\_check\_class`
+      where `class` will be the name or abbreviation of the class of
+      equation. The solving helper will be named as `\_class\_solver`.
+      For example: for exponential equation it becomes `\_check\_expo`
+      and `\_expo\_solver`.
+    - Be sure to consider consider corner cases.
+    - Adding tests for each helper.
+    - Adding a docstring to your helper that describes the method
+      implemented with its proof of correctness by explaining with few
+      examples. Since most of the methods devised to identify and solve
+      particular class of equations are heuristics therefore it
+      becomes necessary to provide proof of correctness of these methods.
+      Following things needs to be included while writing the
+      documentation for the helpers:
+
+      - What is the purpose of the helper.
+      - How it solves the equation and how does it identifies the equation.
+      - Examples should be included to prove its correctness.
+      - What are the input parameters and what does the helper returns.
+    """
+
+    # invert_complex handles the call to the desired inverter based
+    # on the domain specified.
+    lhs, rhs_s = invert_complex(f, 0, symbol, domain)
+
+    result = ConditionSet(symbol, Eq(f, 0), domain)
```

Comment:
You can also write this as:
```
unsolved_result = ConditionSet(symbol, Eq(f, 0), domain)
result = unsolved_result
```
and use the `unsolved_result` in Line 1330, rather than duplicating the same thing.